qiskit/algorithms/minimum_eigen_solvers/adaptvqe.py

```diff
@@ -0,0 +1,321 @@
+        operator: OperatorBase,
+    ) -> List[Tuple[float, PauliSumOp]]:
+        """
+        Computes the gradients for all available excitation operators.
+
+        Args:
+            theta: List of (up to now) optimal parameters
+            vqe: The variational quantum eigensolver instance used for solving
+
+        Returns:
+            List of pairs consisting of gradient and excitation operator.
+        """
+        res = []
+        # compute gradients for all excitation in operator pool
+        sampler = CircuitSampler(self.quantum_instance)
+        for exc in self._excitation_pool:
+            # add next excitation to ansatz
+            self._tmp_ansatz.operators = self._excitation_list + [exc]
+            # the ansatz needs to be decomposed for the gradient to work
+            self.ansatz = self._tmp_ansatz.decompose()
+            param_sets = list(self.ansatz.parameters)
+            # zip will only iterate the length of the shorter list
+            theta1 = dict(zip(self.ansatz.parameters, theta))
+            op, expectation = self.construct_expectation(theta1, operator, return_expectation=True)
+            # compute gradient
+            state_grad = self.gradient.convert(operator=op, params=param_sets)
+            # Assign the parameters and evaluate the gradient
+            value_dict = {param_sets[-1]: 0.0}
+            state_grad_result = sampler.convert(state_grad, params=value_dict).eval()
+            logger.info("Gradient computed : %s", str(state_grad_result))
+            res.append((np.abs(state_grad_result[-1]), exc))
+        return res, expectation
+
+    @staticmethod
+    def _check_cyclicity(indices: List[int]) -> bool:
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: The list of chosen gradient indices.
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    @staticmethod
+    def _log_gradients(
+        iteration: int,
+        cur_grads: Tuple[float, PauliSumOp],
+        max_grad: Tuple[float, PauliSumOp | None],
+    ):
+        if logger.isEnabledFor(logging.INFO):
+            gradlog = f"\nGradients in iteration #{str(iteration)}"
+            gradlog += "\nID: Excitation Operator: Gradient  <(*) maximum>"
+            for i, grad in enumerate(cur_grads):
+                gradlog += f"\n{str(i)}: {str(grad[1])}: {str(grad[0])}"
+                if grad[1] == max_grad[1]:
+                    gradlog += "\t(*)"
+            logger.info(gradlog)
+        else:
+            return
+
+    def compute_minimum_eigensolver(
```

Comment:
This should be `compute_minimum_eigenvalue` so this complies with the MinimumEigenSolver interface. This will end up overriding the one in VQE that can of course still be used from here.

One thought. Does this algorithm need to inherit from VQE or could it be composed with VQE, i.e. take an instance of VQE that it uses at each step. This would open the door to possibly allowing this to use the VQEClient, i,e, the runtime VQE.